api.go

```diff
@@ -479,7 +479,10 @@ func (p *Progress) sizeBytes() uint64 {
 // SearchResult contains search matches and extra data
 type SearchResult struct {
 	Stats
-	Progress
+
+	// Do not encode this as we cannot encode -Inf in JSON
+	Progress `json:"-"`
```

Comment:
@stefanhengl I'm assuming this fix is fine since we use gob in Sourcegraph for reading progress. This PR LGTM but maybe we can do something other than -Inf to make the API a little more ergonomic. Maybe Progress can be optional rather?